src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/ImporterIntegrationTest.java

```diff
@@ -90,6 +90,126 @@ public void createH2Store() throws Exception {
         listener.setCallBack(null);
     }
 
+    @Test
+    public void testDirectWrongFile() throws Exception {
+
+        boolean async = false;
+        // set a callback to check that the request spring context is passed to the job thread
+        RequestContextListener listener = applicationContext.getBean(RequestContextListener.class);
+        SecurityContextHolder.getContext().setAuthentication(createAuthentication());
+
+        final boolean[] invoked = {false};
+        listener.setCallBack(
+                (request, user, resource) -> {
+                    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+                    assertThat(request, notNullValue());
+                    assertThat(resource, notNullValue());
+                    assertThat(auth, notNullValue());
+                    invoked[0] = true;
+                });
+        File dir = unpack("geotiff/EmissiveCampania.tif.bz2");
+        File geotiffFile = new File(dir, "EmissiveCampania.tif");
+        String wrongPath = jsonSafePath(geotiffFile).replace("/EmissiveCampania", "/foobar");
+
+        String wsName = getCatalog().getDefaultWorkspace().getName();
+
+        // @formatter:off
+        String contextDefinition =
+                "{\n"
+                        + "   \"import\": {\n"
+                        + "      \"targetWorkspace\": {\n"
+                        + "         \"workspace\": {\n"
+                        + "            \"name\": \""
+                        + wsName
+                        + "\"\n"
+                        + "         }\n"
+                        + "      },\n"
+                        + "      \"data\": {\n"
+                        + "        \"type\": \"file\",\n"
+                        + "        \"file\": \""
+                        + wrongPath
+                        + "\"\n"
+                        + "      },"
+                        + "      targetStore: {\n"
+                        + "        dataStore: {\n"
+                        + "        name: \"h2\",\n"
+                        + "        }\n"
+                        + "      }\n"
+                        + "   }\n"
+                        + "}";
+        // @formatter:on
+
+        // initialize the import
+        MockHttpServletResponse servletResponse =
+                postAsServletResponse(
+                        "/rest/imports?async=true", contextDefinition, "application/json");
```

Comment:
Use of `async=true` causes [GEOS-8818](https://osgeo-org.atlassian.net/browse/GEOS-8818), fixed in https://github.com/geoserver/geoserver/pull/2962.